orderer/common/multichannel/registrar.go

```diff
@@ -581,11 +581,42 @@ func (r *Registrar) joinAsFollower(ledgerRes *ledgerResources, clusterConsenter
 	return info, nil
 }
 
+type externalHalter interface {
+	HaltExternal()
+}
+
 // RemoveChannel instructs the orderer to remove a channel.
 // Depending on the removeStorage parameter, the storage resources are either removed or archived.
 func (r *Registrar) RemoveChannel(channelID string, removeStorage bool) error {
+	r.lock.RLock()
+	defer r.lock.RUnlock()
+
 	if r.SystemChannelID() != "" && channelID != r.SystemChannelID() {
 		return types.ErrSystemChannelExists
 	}
-	return errors.New("Not implemented yet")
+
+	cs, ok := r.chains[channelID]
+	if !ok {
+		return types.ErrChannelNotExist
+	}
+
+	// check if chain is etcdraft by looking for the HaltExternal function, which
+	// is only implemented by raft chains
+	raftChain, ok := cs.Chain.(externalHalter)
+	if !ok {
+		return errors.Errorf("cannot remove non-raft application channel %s", channelID)
+	}
+
+	raftChain.HaltExternal()
```

Comment:
Not sure we need `HaltExternal`, I think a simple fix in the raft chain can let us use `Halt`.